tests/test_awaitable_task.py

```diff
@@ -0,0 +1,89 @@
+from aiida_workgraph import WorkGraph, task
+import asyncio
+from aiida.cmdline.utils.common import get_workchain_report
+import datetime
+import os
+import pytest
+
+
+def test_awaitable_task(decorated_add):
+    @task.awaitable()
+    async def awaitable_func(x, y):
+        n = 2
+        while n > 0:
+            n -= 1
+            await asyncio.sleep(0.5)
+        return x + y
+
+    wg = WorkGraph(name="test_awaitable")
+    awaitable1 = wg.add_task(awaitable_func, "awaitable_func1", x=1, y=2)
+    add1 = wg.add_task(decorated_add, "add1", x=1, y=awaitable1.outputs["result"])
+    wg.run()
+    report = get_workchain_report(wg.process, "REPORT")
+    assert "Waiting for child processes: awaitable_func1" in report
+    assert add1.outputs["result"].value == 4
```

Comment:
more expressive names are always nicer to read
```suggestion
    awaitable_func_task = wg.add_task(awaitable_func, "awaitable_func_task", x=1, y=2)
    add_task = wg.add_task(decorated_add, "add_task", x=1, y=awaitable_task.outputs["result"])
    wg.run()
    report = get_workchain_report(wg.process, "REPORT")
    assert "Waiting for child processes: awaitable_func_task" in report
    assert add_task.outputs["result"].value == 4
```

Reply:
Thanks! I renamed `awaitable1` to `awaitable_func1`. Generally, I recommend not using `_task` in the task's name, because
- sometimes, the task name is already very long, so we just skip `_task.`
- In the GUI, it will show the name of the task, we try to avoid showing `_task` on the GUI. because we know it's a task.
- A `suffix` is recommend to add to the name, because, we many create multiple tasks from the same Task class.